beacon_chain/spec/eth2_apis/rest_remote_signer_calls.nim

```
@@ -0,0 +1,184 @@
+                    body: Web3SignerRequest): RestPlainResponse {.
+     rest, endpoint: "/api/v1/eth2/sign/{identifier}", meth: MethodPost.}
+  # https://consensys.github.io/web3signer/web3signer-eth2.html#tag/Signing
+
+proc signData*(client: RestClientRef, identifier: ValidatorPubKey,
+               body: Web3SignerRequest
+              ): Future[Web3SignerDataResponse] {.async.} =
+  let startSignTick = Moment.now()
+  inc(nbc_remote_signer_requests)
+  let response =
+    try:
+      await client.signDataPlain(identifier, body)
+    except RestError as exc:
+      let msg = "[" & $exc.name & "] " & $exc.msg
+      debug "Error occured while generating signature",
+            validator = shortLog(identifier),
+            remote_signer = $client.address.getUri(),
+            error_name = $exc.name, error_msg = $exc.msg,
+            signDur = Moment.now() - startSignTick
+      inc(nbc_remote_signer_communication_errors)
+      return Web3SignerDataResponse.err(msg)
+    except CatchableError as exc:
+      let signDur = Moment.now() - startSignTick
+      let msg = "[" & $exc.name & "] " & $exc.msg
+      debug "Unexpected error occured while generating signature",
+            validator = shortLog(identifier),
+            remote_signer = $client.address.getUri(),
+            error_name = $exc.name, error_msg = $exc.msg,
+            signDur = Moment.now() - startSignTick
+      inc(nbc_remote_signer_communication_errors)
+      return Web3SignerDataResponse.err(msg)
+
+  let res =
+    case response.status
+    of 200:
+      inc(nbc_remote_signer_200_responses)
+      let res = decodeBytes(Web3SignerSignatureResponse, response.data,
+                            response.contentType)
+      if res.isErr():
+        let msg = "Unable to decode remote signer response [" &
+                  $res.error() & "]"
+        inc(nbc_remote_signer_failures)
+        return Web3SignerDataResponse.err(msg)
+      let sig = res.get().signature.load()
```

I think it makes sense to verify that the signature is valid here (against the public key and the data) before we broadcast the message on the network. 

I can imagine some configuration issues that may result in invalid signatures. If this happens, we don't want to get de-scored and kicked out from the network if only some of the validators have invalid configuration. Furthermore, we need to inform the user for the problem instead of failing silently.